WakaTime/Watcher.swift

```diff
@@ -126,19 +126,43 @@ private func observerCallback(
     _ notification: CFString,
     _ refcon: UnsafeMutableRawPointer?
 ) {
+    guard let refcon = refcon else { return }
+
+    let axNotification = AXUIElementNotification.notificationFrom(string: notification as String)
+    let this = Unmanaged<Watcher>.fromOpaque(refcon).takeUnretainedValue()
+
+    switch axNotification {
+        case .selectedTextChanged:
+            guard
+                let currentPath = getCurrentPath(element: element, refcon: refcon),
+                let selectedText = element.selectedText,
+                !selectedText.isEmpty
```

Comment:
We can run the event handler even when selected text is empty, or would this cause problems?

Reply:
This will result in a lot of unnecessary calls. We could save the latest "selectedText" and compare it instead of using the selectedText.isEmpty

Reply:
It should be caught down the line with `shouldSendHeartbeat`. Unless it becomes a performance problem, or the currently focused file isn't accurate then it doesn't matter if we call `handleEvent` on every selection change.